ql/ql/src/codeql_ql/ast/Ast.qll

```diff
@@ -983,6 +983,8 @@ class NewTypeBranch extends TNewTypeBranch, Predicate, TypeDeclaration {
 
   override NewTypeBranchType getReturnType() { result.getDeclaration() = this }
 
+  override Annotation getAnAnnotation() { toQL(this).getAFieldOrChild() = toQL(result) }
```

Comment:
Should `getAChild` be updated, or do we not count annotations as children of the nodes they are attached to?

Reply:
The base definition of `getAChild` (in `AstNode`) includes `getAnAnnotation()`.   

So `getAnAnnotation()` is part of `getAChild` here in `NewTypeBranch` through the super call.  

I think that answers your question.

Reply:
Makes sense, thank you!